mne_bids/utils.py

```diff
@@ -26,9 +27,166 @@
+    will generate the *prefix* for a BIDS filename that can be used with many
+    subsequent files, or you may also give a suffix that will then complete
+    the file name.
+
+    Note that not all parameters are applicable to each kind of data. For
+    example, electrode location TSV files do not need a "task" field.
+
+    Parameters
+    ----------
+    subject : str | None
+        The subject ID. Corresponds to "sub".
+    session : str | None
+        The session for a item. Corresponds to "ses".
+    task : str | None
+        The task for a item. Corresponds to "task".
+    acquisition: str | None
+        The acquisition parameters for the item. Corresponds to "acq".
+    run : int | None
+        The run number for this item. Corresponds to "run".
+    processing : str | None
+        The processing label for this item. Corresponds to "proc".
+    recording : str | None
+        The recording name for this item. Corresponds to "recording".
+    space : str | None
+        The coordinate space for an anatomical file. Corresponds to "space".
+    prefix : str | None
+        The prefix for the filename to be created. E.g., a path to the folder
+        in which you wish to create a file with this name.
+    suffix : str | None
+        The suffix for the filename to be created. E.g., 'audio.wav'.
+
+    Examples
+    --------
+    >>> bids_basename = make_bids_basename(subject='test', session='two',
+                                           task='mytask', suffix='data.csv')
+    >>> print(bids_basename)
+    sub-test_ses-two_task-mytask_data.csv
+    >>> print(bids_basename.__repr__())
+    BIDSPath (sub-test_ses-two_task-mytask_data.csv)
+    >>> # copy and update multiple entities at once
+    >>> new_basename = bids_basename.copy().update(subject='test2',
+                                                   session='one')
+    >>> print(new_basename)
+    sub-test2_ses-one_task-mytask_data.csv
+    """
+
+    def __init__(self, subject=None, session=None,
+                 task=None, acquisition=None, run=None, processing=None,
+                 recording=None, space=None, prefix=None, suffix=None):
+        self.update(subject=subject, session=session, task=task,
+                    acquisition=acquisition, run=run, processing=processing,
+                    recording=recording, space=space, prefix=prefix,
+                    suffix=suffix)
+
+    @property
+    def entities(self):
+        """Return dictionary of the BIDS entities."""
+        # create an ordered dictionary of all the bids entities
+        entities = OrderedDict()
+        for key in BIDS_PATH_ENTITIES:
+            value = getattr(self, key, None)
+            if value is not None:
+                entities[key] = value
+        return entities
+
+    def __str__(self):
+        """Return the string representation of the path."""
+        basename = _gen_bids_basename(**self.entities)
+        return basename
+
+    def __repr__(self):
```

Comment:
is this tested?

Reply:
I don't think so. Do you know a good way to test __repr__?

Reply:
just call it :)

BIDSPath().__repr__

or 

assert 'blah' in BIDSPath().__repr__

Reply:
Ah okay that's what I ended up figuring heh. Added in now! Lmk if it doesn't work.